src/adapters/output/postgres/product.ts

```diff
@@ -50,14 +51,37 @@ export class ProductRepository implements ProductRepositoryPort {
     }
   }
 
-  async listByCategory(categoryId: number): Promise<Product[]> {
+  async edit(productId: number, product: Partial<Product>): Promise<DefaultResponse> {
     try {
+      const productRepository = AppDataSource.getRepository(Product);
+      const response = await productRepository.update({ id: productId }, product);
+
+      if (response.affected) {
+        return {
+          message: `Product ${productId} has been updated successfully`,
+          status: 200,
+        };
+      }
+
+      return {
+        message: `Product ${productId} was not updated`,
+        status: 404,
```

Comment:
Shouldn't this code containing the status code be in the controller?

Reply:
the controller sends to client the result of service. In the commented code section  the service returns message and status. I think this approach is clearer than the older. WDYT?